### PR TITLE
Add options to enable/disable nuget search.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
test prtest/win/vsi/p3 please

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest prtest/win/vsi/p2 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_prtest_win_vsi_p2/715/
// Retest reason: https://github.com/dotnet/roslyn/issues/11035

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs [hunk 1]
```diff
@@ -10,13 +10,16 @@
...
 using Microsoft.CodeAnalysis.Packaging;
+using Microsoft.CodeAnalysis.SymbolSearch;
 using Moq;
 using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.AddUsing
 {
-    using FixProviderData = Tuple<IPackageInstallerService, IPackageSearchService>;
+    using CodeAnalysis.Shared.Options;
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs]: Nit: Why are these inside the namespace?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_NuGet.cs]: I fixed.

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Packaging;
+using Microsoft.CodeAnalysis.SymbolSearch;
```
Comment (DustinCampbell) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs]: Why is this needed here?

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs [hunk 3]
```diff
@@ -10,6 +9,8 @@
 using Microsoft.CodeAnalysis.Packaging;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Shared.Options;
+using Microsoft.CodeAnalysis.SymbolSearch;
```
Comment (DustinCampbell) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs]: Sort usings

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs]: Fixed.

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.AssemblyReference.cs]: Types moved out of the packaging namespace into this new namespace.

**Comment by Pilchie** [comment]
:+1:

File: src/Features/Core/Portable/Shared/Options/AddImportOptionsProvider.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Shared.Options
+{
+    [ExportOptionProvider, Shared]
+    internal class AddImportOptionsProvider : IOptionProvider
+    {
+        private readonly IEnumerable<IOption> _options = ImmutableArray.Create<IOption>(
```
Comment (DustinCampbell) [inline src/Features/Core/Portable/Shared/Options/AddImportOptionsProvider.cs]: I'm sad if the generic method needs `<IOption>` :smile:

File: src/VisualStudio/CSharp/Impl/CSharpVSResources.resx [hunk 5]
```diff
@@ -411,4 +408,10 @@
   <data name="NewLinesForBracesProperty" xml:space="preserve">
     <value>Place open brace on new line for properties, indexers, and events</value>
   </data>
+  <data name="Option_Suggest_usings_for_types_in_reference_assemblies" xml:space="preserve">
+    <value>Suggest usings for types in reference assemblies</value>
```
Comment (DustinCampbell) [inline src/VisualStudio/CSharp/Impl/CSharpVSResources.resx]: accelerator key?

File: src/VisualStudio/CSharp/Impl/CSharpVSResources.resx [hunk 6]
```diff
@@ -411,4 +408,10 @@
...
+  </data>
+  <data name="Option_Suggest_usings_for_types_in_NuGet_packages" xml:space="preserve">
+    <value>Suggest usings for types in NuGet packages</value>
```
Comment (DustinCampbell) [inline src/VisualStudio/CSharp/Impl/CSharpVSResources.resx]: ditto

File: src/VisualStudio/CSharp/Impl/CSharpVSResources.resx (hunk 5 above)
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/CSharpVSResources.resx]: Good point.

File: src/VisualStudio/CSharp/Impl/Options/CSharpSettingsManagerOptionSerializer.cs [hunk 7]
```diff
@@ -120,7 +122,8 @@ protected override string GetStorageKeyForOption(IOption option)
         protected override bool SupportsOption(IOption option, string languageName)
         {
             if (option == OrganizerOptions.PlaceSystemNamespaceFirst ||
-                option == OrganizerOptions.WarnOnBuildErrors ||
+                option == AddImportOptions.SuggestForTypesInReferenceAssemblies ||
+                option == AddImportOptions.SuggestForTypesInNuGetPackages ||
```
Comment (DustinCampbell) [inline src/VisualStudio/CSharp/Impl/Options/CSharpSettingsManagerOptionSerializer.cs]: Should we rename `OrganizerOptions` or put these together in the same type (`ImportOptions`?)

File: src/VisualStudio/Core/Def/SymbolSearch/IIOService.cs [hunk 8]
```diff
@@ -1,18 +1,13 @@
...
-using System.Threading.Tasks;
 
-namespace Microsoft.VisualStudio.LanguageServices.Packaging
+namespace Microsoft.VisualStudio.LanguageServices.SymbolSearch
 {
     /// <summary>
     /// Used so we can mock out how the search service does IO for testing purposes.
     /// </summary>
-    internal interface IPackageSearchIOService
+    internal interface IIOService
```
Comment (DustinCampbell) [inline src/VisualStudio/Core/Def/SymbolSearch/IIOService.cs]: StutterStutterService? :smile:

File: src/VisualStudio/Core/Def/SymbolSearch/SymbolSearchService.Update.cs [hunk 9]
```diff
@@ -17,8 +17,12 @@
 using Microsoft.Internal.VisualStudio.Shell.Interop;
 using Roslyn.Utilities;
 using static System.FormattableString;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.Options;
+using System.Linq;
+using System.Collections.Immutable;
```
Comment (DustinCampbell) [inline src/VisualStudio/Core/Def/SymbolSearch/SymbolSearchService.Update.cs]: sort usings

File: src/VisualStudio/Core/Def/SymbolSearch/SymbolSearchService.cs [hunk 10]
```diff
@@ -19,8 +19,11 @@
 using Microsoft.VisualStudio.Shell.Settings;
 using static System.FormattableString;
 using VSShell = Microsoft.VisualStudio.Shell;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SymbolSearch;
```
Comment (DustinCampbell) [inline src/VisualStudio/Core/Def/SymbolSearch/SymbolSearchService.cs]: sort usings

File: src/VisualStudio/VisualBasic/Impl/BasicVSResources.resx [hunk 11]
```diff
@@ -204,4 +204,16 @@
   <data name="Option_GoToDefinition" xml:space="preserve">
     <value>Go to Definition</value>
   </data>
+  <data name="Option_Import_Directives" xml:space="preserve">
+    <value>Import Directives</value>
+  </data>
+  <data name="Option_Suggest_imports_for_types_in_NuGet_packages" xml:space="preserve">
+    <value>Suggest imports for types in NuGet packages</value>
```
Comment (DustinCampbell) [inline src/VisualStudio/VisualBasic/Impl/BasicVSResources.resx]: accelerator key?

File: src/VisualStudio/VisualBasic/Impl/BasicVSResources.resx [hunk 12]
```diff
@@ -204,4 +204,16 @@
...
+  </data>
+  <data name="Option_Suggest_imports_for_types_in_reference_assemblies" xml:space="preserve">
+    <value>Suggest imports for types in reference assemblies</value>
```
Comment (DustinCampbell) [inline src/VisualStudio/VisualBasic/Impl/BasicVSResources.resx]: ditto

**Comment by CyrusNajmabadi** [comment]
Test vsi please

**Comment by DustinCampbell** [comment]
👍

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/vsi/p2 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/vsi/p3 please

**Comment by CyrusNajmabadi** [comment]
test prtest/win/vsi/p2 please
